mesonbuild/wrap/wrap.py

```diff
@@ -610,6 +623,29 @@ def apply_patch(self) -> None:
                 raise WrapException(f'patch directory does not exist: {patch_dir}')
             self.copy_tree(src_dir, self.dirname)
 
+    def apply_diff_files(self) -> None:
+        for filename in self.wrap.diff_files:
+            mlog.log(f'Applying diff file "{filename}"')
+            path = Path(self.wrap.filesdir) / filename
+            if not path.exists():
+                raise WrapException(f'Diff file "{path}" does not exist')
+            if PATCH:
+                cmd = [PATCH, '-f', '-p1', '-i', str(path)]
+            elif GIT:
+                # If the `patch` command is not available, fall back to `git
+                # apply`. The `--work-tree` is necessary in case we're inside a
+                # Git repository: by default, Git will try to apply the patch to
+                # the repository root.
+                cmd = [GIT, '--work-tree', '.', 'apply', '-p1', str(path)]
+            else:
+                raise WrapException(
+                    'Missing "patch" or "git" commands to apply diff files')
```

Comment:
Nit: this doesn't need to be split across two lines. We have other lines of similar length in this file and exceeding the 80 character line limit IMO is better as long as it's not *too* long and completely avoids multi-line invocations.

Reply:
OK, reformatted to a single line.